vector/src/main/java/im/vector/app/features/home/room/detail/timeline/factory/MergedHeaderItemFactory.kt

```diff
@@ -78,19 +79,60 @@ class MergedHeaderItemFactory @Inject constructor(
-            // It's the first item before room.create
-            // Collapse all room configuration events
-            buildRoomCreationMergedSummary(currentPosition, items, partialState, event, eventIdToHighlight, requestModelBuild, callback)
-        } else if (!event.canBeMerged() || (nextEvent?.root?.getClearType() == event.root.getClearType() && !addDaySeparator)) {
-            null
-        } else {
-            buildMembershipEventsMergedSummary(currentPosition, items, partialState, event, eventIdToHighlight, requestModelBuild, callback)
+        return when {
+            isStartOfRoomCreationSummary(event, nextEvent) ->
+                buildRoomCreationMergedSummary(currentPosition, items, partialState, event, eventIdToHighlight, requestModelBuild, callback)
+            isStartOfSameTypeEventsSummary(event, nextEvent, addDaySeparator) ->
+                buildSameTypeEventsMergedSummary(currentPosition, items, partialState, event, eventIdToHighlight, requestModelBuild, callback)
+            isStartOfRedactedEventsSummary(event, nextEvent, addDaySeparator) ->
+                buildRedactedEventsMergedSummary(currentPosition, items, partialState, event, eventIdToHighlight, requestModelBuild, callback)
+            else -> null
         }
     }
 
-    private fun buildMembershipEventsMergedSummary(
+    /**
+     * @param event the main timeline event
+     * @param nextEvent is an older event than event
+     */
+    private fun isStartOfRoomCreationSummary(
+            event: TimelineEvent,
+            nextEvent: TimelineEvent?,
+    ): Boolean {
+        // It's the first item before room.create
+        // Collapse all room configuration events
+        return nextEvent?.root?.getClearType() == EventType.STATE_ROOM_CREATE &&
+                event.isRoomConfiguration(nextEvent.root.getClearContent()?.toModel<RoomCreateContent>()?.creator)
+    }
+
+    /**
+     * @param event the main timeline event
+     * @param nextEvent is an older event than event
+     * @param addDaySeparator true to add a day separator
+     */
+    private fun isStartOfSameTypeEventsSummary(
+            event: TimelineEvent,
+            nextEvent: TimelineEvent?,
+            addDaySeparator: Boolean,
+    ): Boolean {
+        return event.root.getClearType() in mergeableEventTypes &&
+                (nextEvent?.root?.getClearType() != event.root.getClearType() || addDaySeparator)
+    }
+
+    /**
+     * @param event the main timeline event
+     * @param nextEvent is an older event than event
+     * @param addDaySeparator true to add a day separator
+     */
+    private fun isStartOfRedactedEventsSummary(
+            event: TimelineEvent,
+            nextEvent: TimelineEvent?,
+            addDaySeparator: Boolean,
+    ): Boolean {
+        return event.root.isRedacted() &&
+                ((nextEvent?.root?.getClearType() != EventType.REDACTION && !nextEvent?.root?.isRedacted().orFalse()) || addDaySeparator)
```

Comment:
This condition does not work for all cases: some redacted events may not be collapsed since the `nextEvent` (older than the current `event`) is of type `EventType.REDACTION`. We should go further in the events until we have a non redacted event which is not of type `EventType.REDACTION`.

Reply:
Fixed in [59632bf](https://github.com/vector-im/element-android/pull/6501/commits/59632bf00ffca58033a5bb2ef7fa4373ccd4ee96)